### PR TITLE
Add support for LX symlinks (0xA000001D) for WSL/MSYS2 compatibility

### PR DESCRIPTION
**Pull request by vagokuln-msft**
## Summary
Adds support for LX symlinks (IO_REPARSE_TAG_LX_SYMLINK, tag 0xA000001D) used by WSL and MSYS2 for native Unix-style symlinks.

## Problem
Docker builds using MSYS2 with native symlinks (MSYS=winsymlinks:native) fail with error "unsupported reparse point a000001d" because go-winio doesn't recognize LX symlink reparse tags.

## Solution
- Added `reparseTagLxSymlink` constant (0xA000001D)
- Implemented decode for LX symlinks (UTF-8 format with 4-byte version prefix)
- Added `IsLxSymlink` field to preserve symlink type through encode/decode cycles
- Implemented encode to recreate LX symlinks in native format

## Testing
- Unit tests for encode/decode round-trip
- Validated with Docker MSYS2 container builds
- Tested docker commit, save/load cycles preserve LX symlinks correctly

### CODE REVIEW

**Comment by vagokuln-msft** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: reparse.go [hunk 1]
```diff
@@ -69,16 +76,54 @@ func DecodeReparsePointData(tag uint32, b []byte) (*ReparsePoint, error) {
...
+}
+
+func encodeLxReparsePoint(rp *ReparsePoint) []byte {
+	// LX symlink: 4-byte version + UTF-8 target
+	version := uint32(2)
+	targetBytes := []byte(rp.Target)
+	dataLength := 4 + len(targetBytes)
+
+	var b bytes.Buffer
+	_ = binary.Write(&b, binary.LittleEndian, uint32(reparseTagLxSymlink))
```
Comment (Kishon-Microsoft) [inline reparse.go:122]: Any specific reason for discarding the errors?

File: reparse.go [hunk 2]
```diff
@@ -69,16 +76,54 @@ func DecodeReparsePointData(tag uint32, b []byte) (*ReparsePoint, error) {
...
 
 func isDriveLetter(c byte) bool {
 	return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z')
 }
 
-// EncodeReparsePoint encodes a Win32 REPARSE_DATA_BUFFER structure describing a symlink or
-// mount point.
+// EncodeReparsePoint encodes a Win32 REPARSE_DATA_BUFFER structure describing a symlink,
+// mount point, or LX symlink.
 func EncodeReparsePoint(rp *ReparsePoint) []byte {
```
Comment (Kishon-Microsoft) [inline reparse.go:106]: Can we guard against passing a nil value?

File: reparse_lx_test.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
+//go:build windows
+
+package winio
+
+import (
+	"testing"
+)
+
+func TestLxSymlinkRoundTrip(t *testing.T) {
```
Comment (Kishon-Microsoft) [inline reparse_lx_test.go:16]: Can we add tests for other scenarios? atleast with empty Target?

File: reparse.go [hunk 4]
```diff
@@ -69,16 +76,54 @@ func DecodeReparsePointData(tag uint32, b []byte) (*ReparsePoint, error) {
...
+	if rp.IsLxSymlink {
+		return encodeLxReparsePoint(rp)
+	}
+	return encodeWindowsReparsePoint(rp)
+}
+
+func encodeLxReparsePoint(rp *ReparsePoint) []byte {
+	// LX symlink: 4-byte version + UTF-8 target
+	version := uint32(2)
```
Comment (Kishon-Microsoft) [inline reparse.go]: Adding something like const lxSymlinkVersion = 2 and then using it here would be clean.

**Comment by vagokuln-msft** [comment]
@msscotb @Kishon-Microsoft, could you please review this change? We hope to get this merged soon as it has been a few weeks since the PR was opened. Thanks!

File: reparse.go (hunk 1 above)
Comment (vagokuln-msft) [reply reparse.go:122]: The errors are intentionally discarded here. Writing to bytes.Buffer never returns an error since it only performs in-memory operations (no I/O failures possible). The explicit _ assignment makes it clear we're intentionally ignoring the return values rather than accidentally forgetting to check them.

This pattern is used consistently throughout this file, you can see the same approach in encodeWindowsReparsePoint() below. It's a common and accepted Go idiom when working with bytes.Buffer.

File: reparse.go (hunk 4 above)
Comment (vagokuln-msft) [reply reparse.go]: Done

File: reparse.go (hunk 2 above)
Comment (vagokuln-msft) [reply reparse.go:106]: Added a check

File: reparse_lx_test.go (hunk 3 above)
Comment (vagokuln-msft) [reply reparse_lx_test.go:16]: Added more tests

**Review by Kishon-Microsoft — APPROVED** [review]

**Comment by vagokuln-msft** [comment]
Thanks for the sign-off. Looks like all the checks have passed. Hope to have the changes merged in without any problems. Thanks!

**Review by AdityaMittal1306 — APPROVED** [review]
